### ISSUE TITLE
SC2319 warns to assign $? to a var, IN an assignment to a var (0.10.0 broke, shellcheck.net not broken)

### ISSUE THREAD

**Issue by mobilemind** [koalaman/shellcheck#2952]
#### For bugs

- SC2319: This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.
- My shellcheck version (`shellcheck --version` or "online"):  version: 0.10.0
- [X] The rule's wiki page does not already cover this - https://www.shellcheck.net/wiki/SC2319
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit
- Works fine using <https://www.shellcheck.net/>

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

…

# COMMDIR - home dir for common utils
COMMDIR="${HOME}/Developer/common"
## read/write access to $PROJDIR is required
[ -d "${COMMDIR}" ] || die "warning: no directory found at- ${COMMDIR}" 0
[ -w "${COMMDIR}" ] \
	|| { ERR="$?" ; die "write access to ${COMMDIR} is required. Error: $ERR" "$ERR" ; }

```

#### Here's what shellcheck currently says:

In common/bin/commfresh.sh line 90:
	|| { ERR="$?" ; die "write access to ${COMMDIR} is required. Error: $ERR" "$ERR" ; }
                  ^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.

#### Here's what I wanted or expected to see:

No warning using local executable, like I see no warning using shellcheck.net.

**Comment by cbf123** [comment]
I think this is a dupe of https://github.com/koalaman/shellcheck/issues/2937

**Comment by mobilemind** [comment]
> I think this is a dupe of #2937

I'd agree.
